### PR TITLE
chore: clean up ESLint in plugins

### PR DESCRIPTION
**Pull request by Tlacenka**
I went through our plugins and remaining packages and cleaned up what I could.

However, especially `nx-plugin` still has a lot of issues in it and once we start working on it, the first thing should be enabling the ESLint rules and cleaning up what is there.

Before:
image

[After](https://quality-metrics-staging.web.app/portal/code-pushup/cli/branch/eslint-plugin-fixes/dashboard):
image

### CODE REVIEW

File: packages/plugin-eslint/src/lib/meta/rules.ts [hunk 1]
```diff
@@ -27,7 +28,7 @@ export async function listRules(
     {
       messages: rulesIds.map(ruleId => ({ ruleId })),
       suppressedMessages: [] as Linter.SuppressedLintMessage[],
-    } as ESLint.LintResult,
+    } as unknown as ESLint.LintResult,
```
Comment (matejchalk) [inline packages/plugin-eslint/src/lib/meta/rules.ts:31]: Really not a fan of `as unknown as T`. It completely ignores any compatibiliy, while for `as T` TS at least checks it's "similar". Maybe adding `as Linter.LintMessage[]` to `messages` would work?

File: packages/plugin-eslint/src/lib/meta/rules.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ export async function listRules(
   patterns: string | string[],
 ): Promise<RuleData[]> {
   const configs = await toArray(patterns).reduce(
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
     async (acc, pattern) => [
       ...(await acc),
       await eslint.calculateConfigForFile(pattern),
```
Comment (matejchalk) [inline packages/plugin-eslint/src/lib/meta/rules.ts:19]:
```suggestion
    async (acc, pattern) => [
      ...(await acc),
      (await eslint.calculateConfigForFile(pattern)) as Linter.Config,
```

**Review by matejchalk — APPROVED** [review]
Looks good overall :slightly_smiling_face:

File: packages/plugin-eslint/src/lib/meta/rules.ts (hunk 1 above)
Comment (Tlacenka) [reply packages/plugin-eslint/src/lib/meta/rules.ts:31]: No, it does not work. The ESLint problem is that you are using `as` as opposed to `satisfies` 🙄 . So I disabled that.

Comment (matejchalk) [reply packages/plugin-eslint/src/lib/meta/rules.ts]: OK, then you picked the best solution :+1:

**Review by matejchalk — APPROVED** [review]

**Pull request merged**